app/models/game.rb

```diff
@@ -14,6 +14,30 @@ def populate!
     populate_right_white_half!
   end
 
+  def white_check?
```

Comment:
This is what I had initially:

```Ruby
  def check?(color)
    black_king = pieces.find_by(type: 'King', color: 'black')
    white_king = pieces.find_by(type: 'King', color: 'white')
    if color == 'white'
      pieces.select { |p| p.color = 'white' }.each do |piece|
        return true if piece.valid_move?(black_king.x_position,
                                         black_king.y_position)
      end
    elsif color == 'black'
      pieces.select { |p| p.color = 'white' }.each do |piece|
        return true if piece.valid_move?(white_king.x_position,
                                         white_king.y_position)
      end
    end
    false
  end
```
It is 14 lines and is too long for Rubocop.

Then I have removed the top 2 variables outside, so had 12 lines in total. Beyond that...I am not sure how to make it shorter.

@chadbaum with the usage of valid_move? - yes, I remember we arrived at it together, when you saw the older method that I had.